hack/jenkins/e2e-runner.sh

```diff
@@ -207,6 +207,7 @@ function setup_gci_vars() {
 if running_in_docker; then
     record_command "${STAGE_PRE}" "download_gcloud" curl -fsSL --retry 3 --keepalive-time 2 -o "${WORKSPACE}/google-cloud-sdk.tar.gz" 'https://dl.google.com/dl/cloudsdk/channels/rapid/google-cloud-sdk.tar.gz'
     install_google_cloud_sdk_tarball "${WORKSPACE}/google-cloud-sdk.tar.gz" /
+    gcloud auth activate-service-account --key-file="${GOOGLE_APPLICATION_CREDENTIALS}"
```

Comment:
The only reason we have this here is because we don't import the gcloud settings into the docker container.

We have so many entry points into test infra (e2e-runner, node e2e-runner, unit tests, build, upload-to-gcs, maybe others I'm forgetting) that I wanted the account activation in only one place for consistency, i.e. Jenkins.